pkg/messaging/grpc_proxy.go

```diff
@@ -74,7 +75,8 @@ func (p *proxy) intercept(ctx context.Context, fullName string) (context.Context
 
 	v := md.Get(diagnostics.GRPCProxyAppIDKey)
 	if len(v) == 0 {
-		return ctx, nil, func() {}, errors.Errorf("failed to proxy request: required metadata %s not found", diagnostics.GRPCProxyAppIDKey)
+		panic(fmt.Sprintf("failed to proxy request: required metadata %s not found", diagnostics.GRPCProxyAppIDKey))
```

Comment:
This is where the failure happens. I changed the return statement into a panic to get the stack trace.